core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -27,6 +29,16 @@
     @ConfigItem(defaultValue = JAR)
     public String type;
 
+    /**
+     * Experimental property that allows for the creation of a 'mutable' application that
+     * can be re-augmented after creation.
+     *
+     * TODO: should this just be a different package type: 'mutable-jar'
+     *
+     */
+    @ConfigProperty(defaultValue = "true")
+    public boolean mutableApplication;
```

Comment:
Is it really costly? I would have imagined this to be always generated

Reply:
It results in a bigger application, as you need the -deployment artifacts (and their deps) in the final output. For a normal launch these are not loaded, so there is no memory usage or boot time overhead, but it does make the final image bigger.

Reply:
should be default false then?

Reply:
Oops, sorry, trying to do too many things at once and missed that.